python/pyspark/sql/connect/client/core.py

```diff
@@ -688,6 +695,14 @@ def enable_reattachable_execute(self) -> "SparkConnectClient":
         self._use_reattachable_execute = True
         return self
 
+    def register_retry_policy(self, policy: RetryPolicy):
+        if policy.name in self._known_retry_policies:
+            raise ValueError("Already known policy")
+        self._known_retry_policies[policy.name] = policy
+
+    def set_retry_policies(self, policies: List[str]):
+        self._retry_policies = [self._known_retry_policies[name] for name in policies]
```

Comment:
do we need the functionality of registering policies that we are not going to be using?
is there a use case for adding and removing policies?
unless there's a good reason to have a separate `register` and `set`, maybe simplify it and just set is enough?

Reply:
My rationale here was that it makes it easier for users to configure retry policies without needing to toss objects around. Policy consists not just of the class but also parameters set on the instance.

For example, suppose you have a client which shipped with policies ["policyA", "policyB", "policyC"], and for some reason you aren't happy with policyB. Than it's easier to configure this in only one call without having to obtain objects for policyA, policyC. Also convenient to add you own PolicyD in the mix.

Reply:
I'm just wondering if any user that is a "power" user enough to work with changing these policies would not be comfortable with tossing these objects around anyway. E.g. if they would want to tweak the parameters that you mention, they would anyway have to deal with the object, instantiate it with different parameters, register it and then set it.
For such a power user, it actually may be cleaner to just work with these objects directly instead of having a two step registration and setting.

But I don't have a strong opinion here, I am also fine with keeping these extra APIs.

Reply:
> they would anyway have to deal with the object
probably not really needs obtaining the actualy object, but rather import statement from somewhere, and a new instance construction

Reply:
I made a change in API which now also allows passing object directly